flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/table/KafkaTableTestUtils.java

```diff
@@ -74,23 +77,40 @@
     }
 
     public static void waitingExpectedResults(
-            String sinkName, List<String> expected, Duration timeout) throws InterruptedException {
-        long now = System.currentTimeMillis();
-        long stop = now + timeout.toMillis();
+            String sinkName, List<String> expected, Duration timeout)
+            throws InterruptedException, TimeoutException {
         Collections.sort(expected);
-        while (System.currentTimeMillis() < stop) {
-            List<String> actual = TestValuesTableFactory.getResults(sinkName);
-            Collections.sort(actual);
-            if (expected.equals(actual)) {
-                return;
-            }
-            Thread.sleep(100);
-        }
+        CommonTestUtils.waitUtil(
+                () -> {
+                    List<String> actual = TestValuesTableFactory.getResults(sinkName);
+                    Collections.sort(actual);
+                    return expected.equals(actual);
+                },
+                timeout,
+                "Can not get the expected result.");
+    }
 
-        // timeout, assert again
-        List<String> actual = TestValuesTableFactory.getResults(sinkName);
-        Collections.sort(actual);
-        assertEquals(expected, actual);
+    public static void checkReturnEmptyResults(String sinkName, Duration timeout)
+            throws InterruptedException, TimeoutException {
+        CommonTestUtils.waitUtil(
+                new Supplier<Boolean>() {
```

Comment:
I am not the biggest fan of anonymous classes including complex logic, can you extract it? WDYT about adding a common helper to `CommonTestUtils` that combines `waitUntil` and your count-based logic?

Reply:
Yes, I do not think waiting and checking the empty result is a good way to test this situation. I have changed the test to write some data after the job started, and check the new data to make sure we read from latest offsets.